src/Og.php

```diff
@@ -249,15 +249,44 @@ public static function getMemberships(AccountInterface $user, array $states = [O
    *   (optional) Array with the state to return. Defaults to active.
    *
    * @return \Drupal\og\Entity\OgMembership|null
-   *   The OgMembership entity, or NULL if the user is not a member of the
-   *   group.
+   *   The OgMembership entity. If the user is not a member of the group, an
+   *   OgMembership entity will be returned in which the user has the
+   *   'non-member' role.
+   *   NULL will be returned if no membership is available that matches the
+   *   passed in $states.
    */
   public static function getMembership(EntityInterface $group, AccountInterface $user, array $states = [OgMembershipInterface::STATE_ACTIVE]) {
     foreach (static::getMemberships($user, $states) as $membership) {
       if ($membership->getGroupEntityType() === $group->getEntityTypeId() && $membership->getGroupId() === $group->id()) {
         return $membership;
       }
     }
+
+    // The user is not a member, return an OgMembership entity in which the user
+    // has the 'non-member' role.
+    if (in_array(OgMembershipInterface::STATE_PENDING, $states)) {
```

Comment:
Sorry, I'm not a big fan of this change, as it will require extra checks every time we're doing `Og::getMembership`, to see that the return value has a membersihp with a reference to a role that shouldn't be saved like this (in fact we should probably have some code to defend us from creating memberships referencing a `non-member` role).

The workflow as I see it, still remains easy if you care about creating memberships, e.g.:

```php
$membership = Og::getMembership($group, $user) ?: Og::createMembership($group, $user);
```